lib/rspec/matchers/built_in/change.rb

```diff
@@ -344,12 +366,29 @@ def value_representation
 
         def perform_change(event_proc)
           @actual_before = evaluate_value_proc
+          @before_hash = @actual_before.hash
+          yield @actual_before if block_given?
+
+          return false unless Proc === event_proc
```

Comment:
not obvious to me why this was added/needed? When is a non-`Proc` passed?

Reply:
oh, it might be `nil` with some invocations, though in that case why not check for that directly?

Reply:
It an be anything, if a user tries to use it as a value matcher instead of a block matcher, e.g.:

``` ruby
expect(3).to change { ... }
expect(:foo).to change { ... }
```

In these cases, `event_proc` will be `3` and `:foo`, respectively.

Anyhow, this check wasn't actually added; it was moved as part of the diff.